chain/indexer/src/streamer/streamer.rs

```diff
@@ -84,39 +92,64 @@ async fn fetch_block_by_height(
+/// It fetches the block and all related parts (chunks, outcomes, state changes etc.)
 /// and returns everything together in one struct
-async fn fetch_block_response(
+async fn build_streamer_message(
     client: &Addr<near_client::ViewClientActor>,
     block: views::BlockView,
-    outcomes_to_get: impl Iterator<Item = types::TransactionOrReceiptId>,
-) -> Result<FetchBlockResponse, FailedToFetchData> {
+) -> Result<StreamerMessage, FailedToFetchData> {
     let chunks = fetch_chunks(&client, &block.chunks).await?;
-    let (outcomes, mut outcomes_to_retry) = fetch_outcomes(&client, outcomes_to_get).await;
 
-    for chunk in &chunks {
-        outcomes_to_retry.extend(chunk.transactions.iter().map(|transaction| {
-            types::TransactionOrReceiptId::Transaction {
-                transaction_hash: transaction.hash.clone(),
-                sender_id: transaction.signer_id.to_string(),
-            }
-        }));
+    let mut local_receipts: Vec<views::ReceiptView> = vec![];
+    let mut outcomes = fetch_outcomes(&client, block.header.hash).await?;
+    let mut indexer_chunks: Vec<IndexerChunkView> = vec![];
 
-        outcomes_to_retry.extend(chunk.receipts.iter().map(|receipt| {
-            types::TransactionOrReceiptId::Receipt {
-                receipt_id: receipt.receipt_id,
-                receiver_id: receipt.receiver_id.to_string(),
-            }
-        }));
+    for chunk in &chunks {
+        let indexer_transactions = chunk
+            .transactions
+            .iter()
+            .map(|transaction| {
+                if transaction.signer_id == transaction.receiver_id {
+                    // This transaction generates local receipt (sir)
+                    local_receipts.push(views::ReceiptView {
+                        predecessor_id: transaction.signer_id.clone(),
+                        receiver_id: transaction.receiver_id.clone(),
+                        receipt_id: outcomes
+                            .get(&transaction.hash)
+                            .unwrap()
+                            .outcome
+                            .receipt_ids
+                            .first()
+                            .unwrap()
+                            .clone(),
+                        receipt: views::ReceiptEnumView::Action {
+                            signer_id: transaction.signer_id.clone(),
+                            signer_public_key: transaction.public_key.clone(),
+                            gas_price: block.header.gas_price, // TODO: fill it
+                            output_data_receivers: vec![],
+                            input_data_ids: vec![],
+                            actions: transaction.actions.clone(),
+                        },
+                    })
+                }
+                IndexerTransactionWithOutcome {
+                    transaction: transaction.clone(),
+                    outcome: outcomes.remove_entry(&transaction.hash).unwrap().1,
```

Comment:
1. Use `.expect()` and put the reason why it is safe to unwrap here (the transaction execution outcome should always present in the same block as the transaction itself).
2. Use `.remove()` instead of `.remove_entry()` to avoid the need to have `.1`